src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Reflection/Augments/ReflectionAugments.cs

```diff
@@ -82,13 +82,13 @@ public static TypeCode GetRuntimeTypeCode(Type type)
                     break;
             }
 
-            if (type.Equals(CommonRuntimeTypes.String))
+            if (type.Equals(typeof(string)))
```

Comment:
```suggestion
            if (type == typeof(string))
```
When we know that `type` is runtime type, it is more efficient to just use `==` instead of type.Equals.

Reply:
Ok, I see that the other places are using `object.ReferenceEquals(type, typeof(string))`. It works too - it may be better to use that for consistency.

Both `==` and `object.ReferenceEquals` produce same code with optimization on in this case.

Reply:
Opened #621